storm-core/src/jvm/backtype/storm/blobstore/BlobStore.java

```diff
@@ -0,0 +1,446 @@
+   */
+  public abstract AtomicOutputStream createBlob(String key, SettableBlobMeta meta, Subject who) throws AuthorizationException, KeyAlreadyExistsException;
+
+  /**
+   * Updates the blob data.
+   * @param key Key for the blob.
+   * @param who Is the subject creating the blob.
+   * @return AtomicOutputStream returns a stream into which the data
+   * can be written into.
+   * @throws AuthorizationException
+   * @throws KeyNotFoundException
+   */
+  public abstract AtomicOutputStream updateBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException;
+
+  /**
+   * Gets the current version of metadata for a blob
+   * to be viewed by the user or downloaded by the supervisor.
+   * @param key Key for the blob.
+   * @param who Is the subject creating the blob.
+   * @return AtomicOutputStream returns a stream into which the data
+   * can be written into.
+   * @throws AuthorizationException
+   * @throws KeyNotFoundException
+   */
+  public abstract ReadableBlobMeta getBlobMeta(String key, Subject who) throws AuthorizationException, KeyNotFoundException;
+
+  /**
+   * Sets the metadata with renewed acls for the blob.
+   * @param key Key for the blob.
+   * @param meta Metadata which contains the updated
+   * acls information.
+   * @param who Is the subject creating the blob.
+   * @throws AuthorizationException
+   * @throws KeyNotFoundException
+   */
+  public abstract void setBlobMeta(String key, SettableBlobMeta meta, Subject who) throws AuthorizationException, KeyNotFoundException;
+
+  /**
+   * Deletes the blob data and metadata.
+   * @param key Key for the blob.
+   * @param who Is the subject creating the blob.
+   * @throws AuthorizationException
+   * @throws KeyNotFoundException
+   */
+  public abstract void deleteBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException;
+
+  /**
+   * Gets the current version of metadata for a blob
+   * to be viewed by the user or downloaded by the supervisor.
+   * @param key Key for the blob.
+   * @param who Is the subject creating the blob.
+   * @return InputStreamWithMeta has the additional
+   * file length and version information.
+   * @throws AuthorizationException
+   * @throws KeyNotFoundException
+   */
+  public abstract InputStreamWithMeta getBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException;
+
+  /**
+   * Returns an iterator with all the list of
+   * keys currently available on the blob store.
+   * @param who Is the subject creating the blob.
+   * @return Iterator<String>
+   */
+  public abstract Iterator<String> listKeys(Subject who);
+
+  /**
+   * Gets the replication factor of the blob.
+   * @param key Key for the blob.
+   * @param who Is the subject creating the blob.
+   * @return BlobReplication object containing the
+   * replication factor for the blob.
+   * @throws Exception
+   */
+  public abstract int getBlobReplication(String key, Subject who) throws Exception;
```

Comment:
Using a zookeeper call so it throws an Exception. As Exception encompasses AuthorizationException not throwing it